### PR TITLE
Merge stable branch back to master

### PR DESCRIPTION
**Pull request by mrunge**

### CODE REVIEW

File: src/daemon/configfile.c [hunk 1]
```diff
@@ -317,6 +317,38 @@ static int dispatch_loadplugin(oconfig_item_t *ci) {
   return ret_val;
 } /* int dispatch_value_loadplugin */
 
+static int dispatch_value_plugin(const char *plugin, oconfig_item_t *ci) {
```
Comment (rubenk) [inline src/daemon/configfile.c]: You're reintroducing this function, which was removed by d0e2590eec07925abdc656cadca205738e3471ed. Is this by accident?

File: version-gen.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-DEFAULT_VERSION="5.8.1.git"
+DEFAULT_VERSION="5.9.0.git"
```
Comment (rubenk) [inline version-gen.sh]: We're at 5.9.2 now.

File: src/utils/common/common.c [hunk 3]
```diff
@@ -99,10 +99,7 @@ int ssnprintf(char *str, size_t sz, const char *format, ...) {
 
   va_end(ap);
 
-  if (ret < 0) {
-    return ret;
-  }
-  return (size_t)ret >= sz;
+  return ret;
```
Comment (rubenk) [inline src/utils/common/common.c]: This merge is wrong too.

**Comment by rubenk** [comment]
I'm sorry but I can't approve this. Please check all the conflicts again when you merge.

File: src/daemon/configfile.c (hunk 1 above)
Comment (mrunge) [reply src/daemon/configfile.c]: yes, totally by accident.

File: configure.ac [hunk 4]
```diff
@@ -748,7 +748,11 @@ AC_CHECK_FUNCS_ONCE([ \
     if_indextoname \
     setenv \
     setgroups \
-    setlocale
+    setlocale \
```
[the remark this answers is not in the document]
Comment (mrunge) [reply configure.ac]: yes, you are completely right.

File: src/utils/common/common.c (hunk 3 above)
Comment (mrunge) [reply src/utils/common/common.c]: This merge is https://github.com/collectd/collectd/issues/3232 aka https://github.com/collectd/collectd/commit/7a5f35e655e8ba8ccd08fd44bea3574dffa47a5e

Comment (rubenk) [reply src/utils/common/common.c]: Ah yes you're right.

**Comment by rubenk** [comment]
Thanks Matthias!

**Comment by mrunge** [comment]
Thank you Ruben!

**Pull request merged**